format/ComputeIR.fbs

```diff
@@ -0,0 +1,348 @@
+  Inner,
+  LeftOuter,
+  RightOuter,
+  FullOuter,
+  Cross,
+}
+
+table CanonicalJoinKind {
+  id: CanonicalJoinKindId;
+}
+
+table NonCanonicalJoinKind {
+  name_space: string (required);
+  name: string (required);
+}
+
+union JoinKind {
+  CanonicalJoinKind, NonCanonicalJoinKind
+}
+
+/// The contents of Relation.options will be JoinOptions
+/// if Relation.operation = CanonicalOperation::Join
+table JoinOptions {
+  /// The expression which will be evaluated against rows from each
+  /// input to determine whether they should be included in the
+  /// join relation's output.
+  on_expression: Expression (required);
+  /// The kind of join to use.
+  join_kind: JoinKind (required);
+}
+
+/// Whether lesser values should precede greater or vice versa,
+/// also whether nulls should preced or follow values.
+enum Ordering : uint8 {
+  ASCENDING_THEN_NULLS,
+  DESCENDING_THEN_NULLS,
+  NULLS_THEN_ASCENDING,
+  NULLS_THEN_DESCENDING
+}
+
+table SortKey {
+  value: Expression (required);
+  ordering: Ordering = ASCENDING_THEN_NULLS;
+}
+
+/// The contents of Relation.options will be OrderByOptions
+/// if Relation.operation = CanonicalOperation::OrderBy
+table OrderByOptions {
+  /// Define sort order for rows of output.
+  /// Keys with higher precedence are ordered ahead of other keys.
+  keys: [SortKey] (required);
+}
+
+/// The contents of Relation.options will be LimitOptions
+/// if Relation.operation = CanonicalOperation::Limit
+table LimitOptions {
+  /// Set the maximum number of rows of output.
+  count: long;
+}
+
+/// The contents of Relation.options will be CommonOptions
+/// if Relation.operation = CanonicalOperation::Common
+table CommonOptions {
+  /// Commons (CTEs in SQL) allow assigning a name to a stream
+  /// of data and reusing it, potentially multiple times and
+  /// potentially recursively.
+  name: string (required);
+}
+
+/// The contents of Relation.options will be UnionOptions
+/// if Relation.operation = CanonicalOperation::Union
+table UnionOptions {
+  /// For simplicity, all rows from any input to a union relation
+  /// will always be concatenated into a single output- establishing
+  /// uniqueness of output rows is deferred to other relations.
+}
+
+/// The contents of Relation.options will be LiteralOptions
+/// if Relation.operation = CanonicalOperation::Literal
+table LiteralOptions {
+  /// The columns of this literal relation.
+  columns: [Literal] (required);
+}
+
+/// A specification of a query.
+table Plan {
+  /// One or more output relations.
+  sinks: [Relation] (required);
+
+  /// If this Plan was derived from another (for example by running
+  /// an optimization pass), that plan may be included here to
+  /// provide a backtrace of derivations.
+  derived_from: Plan;
+}
+
+/// Frequently there is an open direct channel between producers
+/// of queries (client) and their consumers (server).
+/// `Interactive` is provided to reify this case:
+rpc_service Interactive {
```

Comment:
I believe anything related to gRPC is out of scope for the specification

Reply:
Yup, this can go in examples to help clarify how `Plan`s can be used.